### PR TITLE
Charts farms

### PR DESCRIPTION
**Pull request by rafinskipg**
- Adds recharts library 
- Adds new TVl metrics from the subdao subgraph
- Displaying charts for TVL / Deposits / Withdraws / Rewards over time
- Refactors the datasources tooltip so we can reuse the datasource modal in other components that dont' accept well a tooltip 

image

image

### CODE REVIEW

File: apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+// tvl: number
+
+export function parseFarmChartData(
+  groupBy: 'hour' | 'day' | 'week',
+  deposits: FarmChange[],
+  withdrawals: FarmChange[],
+  rewards: FarmChange[],
+  tvl: FarmChange[]
+) {
+  // For the last 30 days
```
Comment (torhector2) [inline apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts:20]: One possible improvement could be to add more timeframes: 30 days, 3 months, 6 months, 1 year, All time. No need to be done now ofc, just a suggestion.

File: apps/webapp/src/modules/farms/components/history/RewardsStats.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// Helps group the data from the farmHistory rewards
+function getRewardStats(rewards: FarmChange[]) {
+  let daily = 0n;
+  let monthly = 0n;
+  let yearly = 0n;
+
+  // we need to add the rewards for the last year / month / day
+  const now = new Date();
+  const lastYear = new Date(now.getFullYear() - 1, now.getMonth(), now.getDate());
+  const lastMonth = new Date(now.getFullYear(), now.getMonth() - 1, now.getDate());
```
Comment (torhector2) [inline apps/webapp/src/modules/farms/components/history/RewardsStats.tsx:16]: When you do this in January the result for lastMonth is -1

File: apps/webapp/src/modules/farms/components/history/RewardsStats.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  const yesterday = new Date(now.getFullYear(), now.getMonth(), now.getDate() - 1);
```
Comment (torhector2) [inline apps/webapp/src/modules/farms/components/history/RewardsStats.tsx:17]: Same here when the day is the 1st, the result will be 0.

File: apps/webapp/src/modules/farms/components/FarmDetail.tsx [hunk 4]
```diff
@@ -9,40 +8,58 @@ import { ClaimRewards } from './ClaimRewards';
...
+    chainId,
+    contractAddress: farm.contractAddress as `0x${string}`
+  });
+
   return (
     <Box>
       <Flex>
-        <Box sx={{ width: '30%' }}>
+        <Box sx={{ width: '33%' }}>
           <Card m={4}>
```
Comment (torhector2) [inline apps/webapp/src/modules/farms/components/FarmDetail.tsx:38]: I think the FarmDetail component has grown to much and it's a good moment to move every Card into its own Component. It'll make it easier to read and follow.
```
<ManagePositionCard />
<RewardsCard />
...
```

**Review by torhector2 — CHANGES_REQUESTED** [review]
Everything looks very good, I requested changes mostly because of the date calculations I pointed above, I think they're wrong.

If you think that's correct I'll change the review.

**Comment by tyler17** [comment]
Screen Shot 2023-05-08 at 6 33 49 PM
Also noticed DUXCOIN gets cut off on the Y axis here

File: apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts (hunk 1 above)
Comment (tyler17) [reply apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts:20]: 👍 we can add a TODO comment that says we're currently just grouping by day

**Review by tyler17 — APPROVED** [review]
Looks great to me, just left some minor comments

File: apps/webapp/src/modules/farms/components/history/RewardsStats.tsx (hunk 2 above)
Comment (rafinskipg) [reply apps/webapp/src/modules/farms/components/history/RewardsStats.tsx:16]: changed to a timestamp

File: apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts (hunk 1 above)
Comment (rafinskipg) [reply apps/webapp/src/modules/farms/helpers/parseFarmChartData.ts:20]: updated

File: apps/webapp/src/modules/farms/components/history/DepositsWithdrawalsChart.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    return format(new Date(tickItem.toISOString()), dateFormat);
+  };
+
+  return (
+    <Box>
+      <ResponsiveContainer width={'100%'} height={400}>
+        <BarChart data={data} margin={{ bottom: 30, left: 20, right: 20, top: 10 }}>
+          <XAxis
+            dataKey="date"
+            stroke={'#ADADAD'}
```
[the remark this answers is not in the document]
Comment (rafinskipg) [reply apps/webapp/src/modules/farms/components/history/DepositsWithdrawalsChart.tsx:43]: i think the colors are not definitive, i will ask tiago to provide a new palette once we have the new branding

**Review by torhector2 — APPROVED** [review]

**Pull request merged**